From 4847880135ecff2468b97b664afd7a681cdc7fbb Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Tue, 26 Mar 2013 19:01:05 -0400
Subject: [PATCH 540/904] drm/radeon/dpm/trinity: properly catch errors in dpm
 setup

commit c3efac0d5b728daac457421b5fe1494169457568 upstream

We weren't properly catching errors in dpm_enable()
and dpm_set_power_state().

Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/radeon/trinity_dpm.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/radeon/trinity_dpm.c b/drivers/gpu/drm/radeon/trinity_dpm.c
index b2dc905..fce825e 100644
--- a/drivers/gpu/drm/radeon/trinity_dpm.c
+++ b/drivers/gpu/drm/radeon/trinity_dpm.c
@@ -1062,6 +1062,7 @@ static void trinity_update_requested_ps(struct radeon_device *rdev,
 int trinity_dpm_enable(struct radeon_device *rdev)
 {
 	struct trinity_power_info *pi = trinity_get_pi(rdev);
+	int ret;
 
 	trinity_acquire_mutex(rdev);
 
@@ -1085,7 +1086,11 @@ int trinity_dpm_enable(struct radeon_device *rdev)
 
 	if (rdev->irq.installed &&
 	    r600_is_internal_thermal_sensor(rdev->pm.int_thermal_type)) {
-		trinity_set_thermal_temperature_range(rdev, R600_TEMP_RANGE_MIN, R600_TEMP_RANGE_MAX);
+		ret = trinity_set_thermal_temperature_range(rdev, R600_TEMP_RANGE_MIN, R600_TEMP_RANGE_MAX);
+		if (ret) {
+			trinity_release_mutex(rdev);
+			return ret;
+		}
 		rdev->irq.dpm_thermal = true;
 		radeon_irq_set(rdev);
 	}
-- 
1.8.4.93.g57e4c17

